test-two/src/helpers/logger.js

```diff
@@ -1,3 +1,3 @@
-export const log = async message => {
```

Comment:
JavaScript is a single threaded language, and nowadays you can make use of Workers (for multiple threads), but as long as you don't explicitly create and call the worker, you will be in a single thread environment allocating your frames in the stack as you go deeper your calls. 
There are exceptions, of course, like IO calls, browser or node, are made in parallel and returned in a callback and we usually envelop in a promise.

When using async in the code, we end up allocating the frame reference and a couple of layers from the stack in, in the event loop with a good-to-execute-after time.
This function will be called when the main js thread stops its execution, being that with an await or just simply because the code finished. 
All of that is governed (depending by the transpiler you are using) by a regenerator runtime of a kind, babel per instance uses  [facebook's Regenerator](https://github.com/facebook/regenerator).
if you go to babeljs and click the ['try it out'](https://babeljs.io/repl#?browsers=defaults%2C%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYewdgzgLgBBIFsCmAxArmYUCW4YF4YAKASgID4YBvAXwCh5l1MdxSBuIA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Ctypescript&prettier=false&targets=&version=7.22.11&externalPlugins=&assumptions=%7B%7D), check difference of the generated code:
for this
```ts
const someFunction = () => {}
someFunction();
```
and this
```ts
const someFunction = async() => {}
await someFunction();
```

The extra complexity in the calls adds up overtime, when you consider the Big O notation (for both space and time), you end up realizing that the code is not performing as well as it should.

This is my opinion, so that and a screwdriver in my hands is worth a screwdriver, is that there is no such thing as a free lunch. JavaScript is a language that has some really ingenious people, and is pretty complex and interesting in its own rights. When choosing to use a feature is important to know of the trade-offs that it brings to the table.